src/hotspot/share/compiler/abstractCompiler.hpp

```diff
@@ -151,6 +151,8 @@ class AbstractCompiler : public CHeapObj<mtCompiler> {
   bool is_jvmci() const                  { return _type == compiler_jvmci; }
   CompilerType type() const              { return _type; }
 
+  virtual bool is_hidden_from_external_view() const { return false; }
```

Comment:
It would be nice if this had a comment explaining what "hidden from external view" implies. But I see that `Thread::is_hidden_from_external_view` has no documentation either so I guess there's no much that can be explained here if the broader concept is somewhat undefined.

Reply:
I added a comment in CompilerThread.